### ISSUE TITLE
libmemcached failed to build on Ubuntu 14.04.5 LTS

### ISSUE THREAD

**Issue by leo108** [Linuxbrew/homebrew-core#1052]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

https://gist.github.com/leo108/954d269ecc77e66d1de2c3322ddd0f1b

I'm using liunxbrew on an OpenVZ virtual machine

If I run `brew install libmemcached`, it says:

> Last 15 lines from /home/www/.cache/Homebrew/Logs/libmemcached/01.configure:
> 2016-10-14 01:43:40 -0400
>
> ./configure
> --prefix=/home/www/.linuxbrew/Cellar/libmemcached/1.0.18_1
> 
> checking whether the C compiler works... no
> configure: error: in `/tmp/libmemcached-20161014-5495-aupq6v/libmemcached-1.0.18':
> configure: error: C compiler cannot create executables
> See `config.log' for more details
 
I have tried to install libmemcached by interactive mode `brew install --interactive libmemcached`, and run `./configure --prefix=/home/www/.linuxbrew/Cellar/libmemcached/1.0.18_1`, no errors now, and then I input `exit`, it says:

> Error: No such file or directory - /home/www/.cache/Homebrew/Logs/libmemcached/config.log

**Comment by sjackman** [comment]
```
configure:2992: checking whether the C compiler works
configure:3014: /home/www/.linuxbrew/bin/gcc-5 -Os -w -pipe -march=native -undefined dynamic_lookup -isystem/home/www/.linuxbrew/include -L/home/www/.linuxbrew/lib -Wl,--dynamic-linker=/home/www/.linuxbrew/lib/ld.so -Wl,-rpath,/home/www/.linuxbrew/lib conftest.c  >&5
gcc-5: error: dynamic_lookup: No such file or directory
```
https://gist.github.com/leo108/954d269ecc77e66d1de2c3322ddd0f1b#file-config-log-L75

**Comment by leo108** [comment]
@sjackman I'm sure I have installed gcc (both by apt and brew)

**Comment by sjackman** [comment]
PR #1053 should fix your issue. It's being tested now.

**Comment by leo108** [comment]
Is the PR ready to be merge ?

**Comment by sjackman** [comment]
@rwhogg Is `homebrew/core` frozen ❄️ ?

**Comment by rwhogg** [comment]
@sjackman: it is, but I'll merge now and unfreeze if you're okay with it. There's nothing in this release that looks too scary to me; one of the easiest merges I've done so far.

**Comment by sjackman** [comment]
:+1: Yes, I'm okay with that. Thanks for handling the merge!

**Comment by sjackman** [comment]
Merged. Fixed by #1053

**Comment by sjackman** [comment]
Thanks for the bug report, @leo108
